parser/src/parser/tests/boundary_constraints.rs

```diff
@@ -83,7 +87,7 @@ fn boundary_constraint_with_pub_input() {
             boundary_constraints: vec![BoundaryConstraint::new(
                 Identifier("clk".to_string()),
                 Boundary::First,
-                BoundaryExpr::PubInput(Identifier("a".to_string()), 0),
+                BoundaryExpr::VecElem(Identifier("a".to_string()), 0),
```

Comment:
the name of this test should be updated

Reply:
I went with adding a `public_inputs` section instead and kept the name same.